### ISSUE TITLE
[v4-alpha.2] Event recurrence start/end 

### ISSUE THREAD

**Issue by billyshena** [fullcalendar/fullcalendar#4335]
Hi,

I have recently upgraded to the 4.0-alpha version to use the recurrence feature on the Event model.
When we specify a start/end attribute on the event object, the event does not appear on the calendar.

For example:

```
// Working 
let event = { daysOfWeek: [1,2,3], startTime: '08:00', endTime: '10:00' } 

// Not working 
let event = { daysOfWeek: [1,2,3], startTime: '08:00', endTime: '10:00', start: new Date(), end: new Date() } 

// Rendering event
cal.addEvent(event);

```

Would be great to be able to keep start/end properties since we need to access those attributes when an Event is updated on the calendar.

Is it a bug ? For the moment I will store start/end in the extendedProps

Best,

**Comment by billyshena** [comment]
Another issue I just found, the 'allDay' attribute does not work if I also have 'daysOfWeek' attribute.


```
// Working
cal.addEvent({
    daysOfWeek: [1, 2, 3],
    title: 'Test event',
    startTime: '10:00',
    endTime: '12:00'
  })

// Not working
cal.addEvent({
    daysOfWeek: [1, 2, 3],
    allDay: true,
    title: 'Test event',
    startTime: '10:00',
    endTime: '12:00'
  })
```

**Comment by acerix** [comment]
Here's the docs on recurring events for reference:

https://fullcalendar.io/docs/v4/recurring-events

For your first issue, I believe you just need to replace `start` and `end` with `startRecur` and `endRecur`.

For the second issue, what are you looking for it to do for an event that is allDay and has a start/end times?  By definition, an "all day" event starts goes from the beginning of the day to the end. 

If you can post a [runnable, stripped-down demonstration of the bugs](http://fullcalendar.io/wiki/Reduced-Test-Cases/) it will help us to reproduce.

**Comment by adjioev** [comment]
Hi @acerix,

I've tried to use `startRecur` and `endRecur` props, here is my test event:
````
      events: [
        {
          title: 'Event Title 1',
          daysOfWeek: [1],
          startTime: '02:00',
          endTime: '04:00',
          startRecur: '2018-11-15',
          endRecur: '2018-12-22'
        }
      ]
````

but I get `ERROR TypeError: Cannot read property 'start' of null` error if I open calendar outside event range (startRecur/endRecur).

When ranges don't intersect `intersectRanges` returns null and then we get nullPointException:
````
  let dayMarker = startOfDay(framingRange.start)
````
in `expandRanges` method because framingRange is null.

Thanks Alex

**Comment by acerix** [comment]
Would you be able to post a [runnable, stripped-down demonstration of the bug](http://fullcalendar.io/wiki/Reduced-Test-Cases/)? Would really appreciate it because the time saved reproducing will be time spent fixing.

**Comment by adjioev** [comment]
Hi @acerix , 

I have angular demo  https://github.com/adjioev/fullcalendar-bug

If it's not convinient for you let me know, I will try to set it up in codepan.

Thanks Alex

**Comment by acerix** [comment]
Thanks for the demo, I see the error and it looks like you're using it correctly.

I'm not sure it's the same issue that was originally reported though, in which case it should be a new issue.  @billyshena Are you getting the same error?

**Comment by anoopar** [comment]
Hi @acerix ,
I am facing the same issue as @adjioev .
Is this planned to get fixed, or needs to be filed as seperate issue?
Thanks

**Comment by adjioev** [comment]
@anoopar,

if you need a quick temporary fix you can edit fullcaleadar.js source, look for `expandRanges` method
and add `if (framingRange === null) { return []; }` check:

````
function expandRanges(daysOfWeek, startTime, recurRange, framingRange, dateEnv) {
    framingRange = date_range_1.intersectRanges(framingRange, recurRange);
   if (framingRange === null) { return []; }
    var dowHash = daysOfWeek ? object_1.arrayToHash(daysOfWeek) : null;
    var dayMarker = marker_1.startOfDay(framingRange.start);
    var endMarker = framingRange.end;
    ....
````
Cheers Alex

**Comment by acerix** [comment]
I just moved that into a new issue because I don't think it's the same as what @billyshena reported.

**Comment by arshaw** [comment]
seems like all user-errors have been addressed and that the real bug ticket is this #4369, so closing this ticket.
